tamagotchi1/tests/hello_world_test.rs

```diff
@@ -1,58 +1,44 @@
-// use gtest::{Log, Program, System};
-//
-// #[test]
-// fn hello_test() {
-//     let sys = System::new();
-//     sys.init_logger();
-//     let program = Program::current(&sys);
-// //Note that we added sys.init_logger() to initialize
-// //printing logs into stdout,
-// //and we sent a message from the user with id 2 (id 2 transforms to ActorId equal to 0x020000…00 ).
-//     program.send(2, String::from("INIT MESSAGE"));
-//
-//     let expected_log = Log::builder()
-//         .dest(2)
-//         .payload(String::from("Hello"));
-//     assert!(res.contains(&expected_log));
-// }
-
-
 use gtest::{Log, Program, System};
-// use hello_world::InputMessages;
-use tamagotchi_io::InputMessages;
+use tamagotchi_io::{TmgAction, TmgEvent};
 
 #[test]
 fn hello_test() {
     let sys = System::new();
     sys.init_logger();
     let program = Program::current(&sys);
+
+    let t_name = "7Tamagotchi42";
+
     // The first message to the Program structure is always the initialization message even if the program does not have the init function.
-    //let res = program.send_bytes(2, String::from("Hello"));
-    let res = program.send(2, String::from("Hello"));
+    let sender_id = 2;
+    let res = program.send(sender_id, String::from(t_name));
     assert!(!res.main_failed());
-    assert!(res.log().is_empty());
-
-    // test `SendHelloTo`
-    let hello_recipient: u64 = 4;
-    let res = program.send(
-        2,
-        InputMessages::SendHelloTo(hello_recipient.into()),
-    );
+    assert!(!res.log().is_empty());
 
+    //test init()
     let expected_log = Log::builder()
-        .dest(hello_recipient)
-        .payload(String::from("Hello"));
+        .dest(sender_id)
+        .payload(String::from("Success!"));
 
-    assert!(res.contains(&expected_log))
-}
+    assert!(res.contains(&expected_log));
 
+    // test for TmgAction::Name
+    let sender_id = 3;
+    let res = program.send(sender_id, TmgAction::Name);
 
-// What is the purpose of the init() function in the smart contract code?
-// +To emulate the node behaviour for running programs
-// -To handle messages received by a program
-// -To create a new program
-// -To handle messages received by a program
+    let expected_log = Log::builder()
+        .dest(sender_id)
+        .payload(TmgEvent::Name(String::from(t_name)));
+
+    assert!(res.contains(&expected_log));
+    assert!(!res.main_failed());
 
+    // test for TmgAction::Age
+    let sender_id = 3;
+    let res = program.send(sender_id, TmgAction::Age);
+
+    assert!(!res.main_failed());
```

Comment:
Just FYI you could check the age of your Tamagotchi in the test by using [`System::spend_blocks`](https://docs.gear.rs/gtest/struct.System.html#method.spend_blocks) function.